lib/screens/authentication/import_key/import_key_screen.dart

```diff
@@ -63,12 +63,11 @@ class ImportKeyScreen extends StatelessWidget {
                                   onPressed: () async {
                                     final clipboardData = await Clipboard.getData('text/plain');
                                     final clipboardText = clipboardData?.text ?? '';
-                                    final newText =
-                                        textController.value.replaced(textController.selection, clipboardText).text;
-                                    textController.text = newText;
+                                    textController.text = clipboardText;
                                     // ignore: use_build_context_synchronously
-                                    BlocProvider.of<ImportKeyBloc>(context)
```

Comment:
this code was preventing the first tap from copying the text into the field.

Reply:
No, you can't do that. 

I am not sure what the problem was with the existing code, but I put that code in there to make copy / paste work correctly. 

Copy paste means that whatever is currently selected in the text field is replaced by the pasted text. 

This is basics, it needs to work this way. 

It's not just about the button but also about hitting "paste" from the text editing overlay. 

For example let's say the text is 

"12345foobar"

I go outside the app and copy the text 6789 to the clipboard. . 

Then I go back and select "foobar", then hit paste - now the new string should 123456789

If we just replace text with clipboard then we get only 6789 as the new text. 

I will check into the first tap issue.

Reply:
It's how any text editor works, so we need to work the same way

It's going to be maybe rare that someone wants to copy paste parts of a string but from a UX perspective ... copy paste rules must be followed.